### DISCUSSION TITLE
Game not starting

### DISCUSSION THREAD

**Discussion by rushkii** [vfsfitvnm/frida-il2cpp-bridge#349] [Q&A]
Do I really need to frida-server running on the Android process? Because when I open the game, the game itself not starting until I kill the frida-server process on the Android.

**Comment by vfsfitvnm** [comment]
It looks like the game is detecting frida - if it doesn't open even if frida isn't attached to the game, it's checking either the default frida port or frida specific files in the directory where your frida-server is located at

**Reply by rushkii** [reply]
update: I used frida gadget using this repo https://github.com/lico-n/ZygiskFrida, but when I tried to attach, it get me an error and then connection terminated.

![image](https://github.com/vfsfitvnm/frida-il2cpp-bridge/assets/47179236/bb506376-e7b4-44f3-91cc-09522aaac5c8)

**Reply by vfsfitvnm** [reply]
Unfortunately I cannot help you, it's clear the game has some anti-debugging mechanism, so you have to investigate on your own. I suggest you to make some extra effort to use `frida-server` to gain faster iteration cycles.

Looking at that error, my take is the developers altered the IL2CPP source code, but again you have to pinpoint the precise line of code and find alternatives.

If you tell me the game name, I *may* take a look myself in the future (assume it's not prioritary)

**Reply by rushkii** [reply]
The game name is **Mobile Legends: Bang Bang** ([Play Store Link](https://play.google.com/store/apps/details?id=com.mobile.legends)). I've tried to enable `frida-server`, but I don't understand why the game won't open when `frida-server` is enabled. I will attempt another solution using `jshook` that has it's own `frida-server`.

Another guy was open a discussion asking the same as the game I tried to ask about, However, I don't know what method he is using.
https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/252#discussion-4801633
